packages/protocol/contracts/test/libs/TestLibProving.sol

```diff
@@ -232,20 +221,19 @@ library TestLibProving {
                     )
                 );
 
-                if (
-                    !proofVerifier.verifyZKP({
-                        verifierId: string(
-                            abi.encodePacked(
-                                "plonk_verifier_",
-                                i,
-                                "_",
-                                evidence.circuits[i]
-                            )
-                        ),
-                        zkproof: evidence.proofs[i],
-                        instance: instance
-                    })
-                ) revert L1_ZKP();
+                bool verified = proofVerifier.verifyZKP({
+                    verifierId: string(
+                        abi.encodePacked(
+                            "full_zkp_verifier_id_",
+                            i,
+                            "_prove_",
+                            evidence.circuits[i]
+                        )
+                    ),
+                    zkproof: evidence.proofs[i],
+                    instance: instance
```

Comment:
Seems like the sig proof isn't part of the circuit input here? Or will be current implementation verify that proof on-chain as well?

Reply:
> Seems like the sig proof isn't part of the circuit input here? Or will be current implementation verify that proof on-chain as well?

I am assuming the hash of the signature proof doesn't have to be part of the input of the main circuit and the main circuit always **duplicates** the verification of all signatures (and other validity rules). Unless you want to the main circuits to verify the signature proof itself instead of having the signature circuits built in. Let me know your preference, Brecht.

Reply:
Ah I see, I think that's a good way to do!

Reply:
Oh wait, there may be a problem with this. The proposer could submit an incorrect proof for a valid block. But then the prover has the option to either prove the block, or verify the incorrect proof and show that the block is invalid. This would make the chain nondeterministic.

Reply:
Then I guess the only way to fix it is to hash the signature zk-proof into the main circuit's input and recursively prove the signature proof.